tools/PI/DevHome.PI/Views/BarWindowHorizontal.xaml.cs

```diff
@@ -408,33 +390,55 @@ public void SetRegionsForTitleBar()
 
     private void SetDefaultPosition()
     {
+        // First set our default size before setting out position
+        SetDefaultWidthAndHeight();
+
         // If attached to an app it should show up on the monitor that the app is on
-        _monitorRect = GetMonitorRectForWindow(_viewModel.ApplicationHwnd ?? TryGetParentProcessHWND() ?? ThisHwnd);
-        var screenWidth = _monitorRect.right - _monitorRect.left;
+        // Be sure to grab the DPI for that monitor
+        var dpiScale = GetDpiScaleForWindow(_viewModel.ApplicationHwnd ?? TryGetParentProcessHWND() ?? ThisHwnd);
+
+        RECT monitorRect = GetMonitorRectForWindow(_viewModel.ApplicationHwnd ?? TryGetParentProcessHWND() ?? ThisHwnd);
+        var screenWidth = monitorRect.right - monitorRect.left;
         this.Move(
-            (int)((screenWidth - (Width * _dpiScale)) / 2) + _monitorRect.left,
-            (int)WindowPositionOffsetY + _monitorRect.top);
+            (int)((screenWidth - (Width * dpiScale)) / 2) + monitorRect.left,
+            (int)WindowPositionOffsetY + monitorRect.top);
+    }
 
+    internal void SetDefaultWidthAndHeight()
+    {
         // Get the saved settings for the ExpandedView size. On first run, this will be
         // the default 0,0, so we'll set the size proportional to the monitor size.
         // Subsequently, it will be whatever size the user sets.
-        var settingSize = Settings.Default.ExpandedLargeSize;
-        if (settingSize.Width == 0)
+        RECT monitorRect = GetMonitorRectForWindow(_viewModel.ApplicationHwnd ?? TryGetParentProcessHWND() ?? ThisHwnd);
+
+        var settingWidth = Settings.Default.WindowWidth;
+        if (settingWidth == 0)
         {
-            settingSize.Width = (int)((_monitorRect.Width * 2) / (3 * _dpiScale));
+            settingWidth = monitorRect.Width * 2 / 3;
+            Settings.Default.WindowWidth = settingWidth;
         }
 
-        if (settingSize.Height == 0)
+        var settingHeight = Settings.Default.ExpandedWindowHeight;
+        if (settingHeight == 0)
         {
-            settingSize.Height = (int)((_monitorRect.Height * 3) / (4 * _dpiScale));
+            settingHeight = monitorRect.Height * 3 / 4;
+            Settings.Default.ExpandedWindowHeight = settingHeight;
         }
 
-        Settings.Default.ExpandedLargeSize = settingSize;
-        Settings.Default.Save();
+        // Set the default window width
+        Width = Settings.Default.WindowWidth;

```

Comment:
>Width [](http://example.com/codeflow?start=8&length=5)

Width is in dpi-scaled pixels, but you've removed them from the calculations above.

Reply:
Gah, you're right. As I was playing with this, I realized I was adjusting more widths/heights than I needed to. However, in this case, when you're manipulating the monitor properties (that doesn't have DPI awareness) you need to apply the DPI adjustment.